### PR TITLE
Block REFRESH MATERIALIZED VIEW on caggs

### PR DESCRIPTION
**Pull request by svenklemm**

### CODE REVIEW

File: src/process_utility.c [hunk 1]
```diff
@@ -3396,24 +3396,21 @@ process_refresh_mat_view_start(ProcessUtilityArgs *args)
...
-	FC_SET_NULL(fcinfo, 1);
-	FC_SET_NULL(fcinfo, 2);
-	ts_cm_functions->continuous_agg_refresh(fcinfo);
+	if (cagg)
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("operation not supported on continuous aggregate"),
+				 errdetail("A continuous aggregate does not support REFRESH MATERIALIZED VIEW."),
+				 errhint("Use \"refresh_continuous_aggregate\" or set up a policy to refresh the "
+						 "continuous aggregate.")));
```
Comment (mkindahl) [inline src/process_utility.c:3411]: No test coverage for this piece of code. Add a test?

File: tsl/test/sql/continuous_aggs_refresh.sql [hunk 2]
```diff
@@ -237,12 +237,6 @@ GROUP BY 1,2 WITH DATA;
 SELECT * FROM weekly_temp_without_data;
 SELECT * FROM weekly_temp_with_data;
 
--- Test that REFRESH MATERIALIZED VIEW works as an alternative to
--- refresh_continuous_aggregate()
-REFRESH MATERIALIZED VIEW weekly_temp_without_data;
```
Comment (mkindahl) [inline tsl/test/sql/continuous_aggs_refresh.sql:242]: You could keep this `REFRESH` and just check that it generates an error.

**Review by mkindahl — APPROVED** [review]

**Review by k-rus — COMMENTED** [review]
I support that there should be a test for calling `REFRESH MATERIALIZED VIEW` and getting the error message.

**Review by pmwkaa — APPROVED** [review]
Like missing the error case as @mkindahl pointed out

**Pull request merged**